lib/components/isometric_tile_map_component.dart

```diff
@@ -0,0 +1,106 @@
+import 'dart:ui';
+
+import 'package:flame/components/component.dart';
+
+import '../flame.dart';
+import '../position.dart';
+import '../sprite.dart';
+
+class IsometricTileset {
+  final Image tileset;
+  final int size;
+
+  final Map<int, Sprite> _spriteCache = {};
```

Comment:
can't we use SpriteSheet here?

Reply:
I'm not super familiar with it, but I can take a look

Reply:
this is probably a good opportunity to make the SpriteSheet class be lazy loaded.

Reply:
it's exactly the same thing hahaha
I will refactor it to use sprite sheet
do you think lazy is better? when I implemented I was over the fence of doing lazy or not. I can do the refactor if you prefer but I am not convinced either way

Reply:
I do, Rafa faced some issues using the SpriteSheet class on bonfire to load big images due to it trying to load all the sprites upon instance creation.

Reply:
I will wait for the refactor to do this change more easily :)